tests/VolleyManagement.Domain.UnitTests/TeamTests.cs

```diff
@@ -0,0 +1,250 @@
+            act.Should().Throw<ArgumentException>("Coach Name can`t contain numbers.");
+        }
+
+        [Fact]
+        public void Team_CoachNameContainsSymbol_ExceptionIsThrown()
+        {
+            //Arrange
+            var team = CreateTeam();
+
+            //Act
+            Action act = () => { team.Coach = "Coah#"; };
+
+            //Assert
+            act.Should().Throw<ArgumentException>("Coach Name can`t contain symbols.");
+        }
+
+        #endregion
+
+        #region Achievements
+
+        [Fact]
+        public void Team_MaxAchievementsLength_ExceptionIsThrown()
+        {
+            //Arrange
+            var team = CreateTeam();
+
+            //Act
+            Action act = () => { team.Achievements = GenerateTeamStrings(Constants.Team.MAX_ACHIEVEMENTS_LENGTH + 1); };
+
+            //Assert
+            act.Should().Throw<ArgumentException>("Achievements length should not be longer than maximum allowed value.");
+        }
+
+        [Fact]
+        public void Team_EmptyAchievements_ExceptionIsNotThrown()
+        {
+            //Arrange
+            var team = CreateTeam();
+
+            //Act
+            Action act = () => { team.Achievements = string.Empty; };
+
+            //Assert
+            act.Should().NotThrow<ArgumentException>("Achievements length can be empty");
+        }
+
+        #endregion
+
+        #region CaptainId
+
+        [Fact]
+        public void Team_EmptyCaptainId_ExceptionIsThrown()
+        {
+            //Arrange
+            var team = CreateTeam();
+
+            //Act
+            Action act = () => { team.CaptainId = null; };
+
+            //Assert
+            act.Should().Throw<ArgumentException>("Create team without Captain not allowed");
+        }
+
+        [Fact]
+        public void Team_ZeroCaptainId_ExceptionIsThrown()
+        {
+            //Arrange
+            var team = CreateTeam();
+
+            //Act
+            Action act = () => { team.CaptainId = new PlayerId() { Id = Constants.Team.MIN_ID - 1 }; };
+
+            //Assert
+            act.Should().Throw<ArgumentException>("Create team without Captain not allowed, CaptainId can`t be less then minimum alloved value");
+        }
+
+        #endregion
+
+        #region List<PlayerId>
+
+        [Fact]
+        public void Team_RosterIsNull_ExceptionIsThrown()
+        {
+            //Arrange
+            var team = CreateTeam();
+
+            //Act
+            Action act = () => { team.Roster = null; };
+
+            //Assert
+            act.Should().Throw<ArgumentException>("Team Roster can`t be null.");
+        }
+
+        [Fact]
+        public void Team_MinPlayerID_ExceptionIsThrown()
+        {
+            //Arrange
+            var team = CreateTeam();
+
+            //Act
+            Action act = () =>
+            {
+                team.Roster = new List<PlayerId>() {
+                new PlayerId() {
+                    Id = Constants.Team.MIN_ID - 1
```

Comment:
Why PlayerId is not immutable?

Reply:
It is now.

Reply:
Which is why it wasn't changed previously.